src/Components/Post/Post-Gallery.js

```diff
@@ -4,99 +4,131 @@ import { connect } from 'react-redux';
 import Actions from "../../Actions/actions";
 import { library } from '@fortawesome/fontawesome-svg-core';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faMapMarkerAlt, faFeather, faPencilAlt, faAngleLeft, faAngleRight} from '@fortawesome/free-solid-svg-icons';
+import { faMapMarkerAlt, faAngleLeft, faAngleRight} from '@fortawesome/free-solid-svg-icons';
 import Constants from '../../constants';
-library.add(faMapMarkerAlt, faFeather, faPencilAlt, faAngleLeft, faAngleRight) 
+library.add(faMapMarkerAlt, faAngleLeft, faAngleRight) 
 
 class PostGallery extends Component {
-    renderEditIcon(){
-        if (!this.props.editable) {
-            return null
+    constructor(props) {
+        super(props);
+        this.state = {
+            showImageNavigationButtons: false,
+            currentDisplayImageSrc: this.setInitialDisplayImageState(),
+        }
+    }
+
+    setInitialDisplayImageState() {
+        if (this.props.images.length === 0) {
+            return Constants.POST_GALLERY_DEFAULT_IMAGE
+        } else {
+            return this.props.images[0]
         }
-        return <FontAwesomeIcon icon="pencil-alt" className="edit_icon" />
     }
 
     handleClickedImage(currentViewingImage, postImages){
+        if (currentViewingImage === (Constants.UPLOADS_HOSTNAME + Constants.DEFAULT_IMG_SRC)) {
+             return null
+        }
         this.props.toggleModal('photos', true);
         this.props.handleClickedImage({currentViewingImage, postImages});
-    }
+}
 
-    handleImages(){
-        if (!this.props.images) {
-            return null
-        }
-        const images = [];
-        let index = 1;
-        for (let image of this.props.images) {
-                images.push(
-                    <img className="image" 
-                            src={image} 
-                            alt=" " 
-                            key={image + this.props.name + index}
-                            onClick={ ()=> this.handleClickedImage(image, this.props.images) }>
-                    </img>
-                )
-                index++;
+    onThumbnailButtonsClicked(isLeft) {
+        const currentIndex = this.props.images.indexOf(this.state.currentDisplayImageSrc);
+        let nextIndex = currentIndex - 1;
+        if (isLeft) {
```

Comment:
This is weirdly asymmetrical. The `else` branch adds 1, but the `if` branch doesn't. More readable if symmetric.